sentry/src/main/java/io/sentry/transport/AsyncHttpTransport.java

```diff
@@ -224,14 +246,29 @@ public void run() {
 
             options.getLogger().log(SentryLevel.ERROR, message);
 
+            // ignore e.g. 429 as we're not the ones actively dropping
+            if (result.getResponseCode() >= 500) {
+              if (!(sentrySdkHint instanceof Retryable)) {
+                clientReportRecorder.recordLostEnvelope(
+                    DiscardReason.NETWORK_ERROR, envelopeWithClientReport, options);
+              } else {
+                clientReportRecorder.recordLostClientReportInEnvelope(
+                    DiscardReason.NETWORK_ERROR, envelopeWithClientReport, options);
+              }
+            }
+
             throw new IllegalStateException(message);
           }
         } catch (IOException e) {
           // Failure due to IO is allowed to retry the event
           if (sentrySdkHint instanceof Retryable) {
             ((Retryable) sentrySdkHint).setRetry(true);
+            clientReportRecorder.recordLostClientReportInEnvelope(
```

Comment:
@marandaneto I've added this to all the places were I did nothing for retryable envelopes before. The envelope will not be written to disk with client report on error so the counts would be lost. Therefore I now restore the counts of retryables without recording the rest of the envelope as lost. Can you please confirm this is the right way to do it?

Reply:
Going to remove this again since `Retryables` are only sent on startup where we won't have any client reports in memory anyways. This check as implemented also would require more work to not add counts from an envelope that was cached to disk including a client report. If that would fail multiple times, the client report counts would be re-added multiple times.

Reply:
One thing to test out is the `EnvelopeFileObserver`, since Events written by hybrid SDKs (Flutter and RN) are actually `Retryable`, I don't think it's gonna be a problem, but good to keep in mind.

Reply:
I think we don't need a special logic for client reports for the first iteration. Just add the client report to the envelope and don't worry about recovering them. Adding a recovery logic also opens up the door for bugs as double counting.